PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift

```diff
@@ -110,32 +105,34 @@ class SearchViewModel: ObservableObject {
     func clear() {
         searchResults = []
         subscriptions = []
+        self.savesLocalSearch.clear()
+        self.savesOnlineSearch.clear()
+        self.archiveOnlineSearch.clear()
+        self.allOnlineSearch.clear()
     }
 
-    func clearCache() {
-        cachedSearchResults = [
-            .saves: [:],
-            .archive: [:],
-            .all: [:]
-        ]
-    }
-
-    private func retrieveCachedResults() -> Bool {
-        if let cachedItems = cachedSearchResults[selectedScope]?[currentSearchTerm], !cachedItems.isEmpty {
-            self.searchResults = cachedItems
-            return true
-        }
-        return false
-    }
-
-    private func submitOnlineSearch(with term: String) {
-        clear()
-        searchService.results.receive(on: DispatchQueue.main).sink { [weak self] items in
-            guard let self else { return }
-            self.searchResults = items.compactMap { SearchItem(item: $0) }
-        }.store(in: &subscriptions)
-        Task {
-            await searchService.search(for: term, scope: selectedScope)
+    private func submitSearch(with term: String) {
+        switch selectedScope {
+        case .saves:
+            // TODO: Handle Offline for Premium https://getpocket.atlassian.net/browse/IN-971
+            guard isPremium else {
+                self.searchResults = savesLocalSearch.search(with: term)
+                return
+            }
+            savesOnlineSearch.search(with: term)
+            savesOnlineSearch.$results.receive(on: DispatchQueue.main).sink { items in
+                self.searchResults = items
```

Comment:
We still want to be weary of setting this _every_ time we receive results. We want to make sure we're setting this value only if the currently selected scope matches that of the new search helper time.